File: KIJUNG/WEEK_05/10828.cpp
#include <iostream>
#include <stack>
#include <string>

#define FOR(i,a,b) for(int i=a;i<b;i++)

using namespace std;

int N;
string str;
stack<int> s;


int main(void){
    cin >> N;

    FOR(i,0,N){
        cin >>str;

        if(str == "push"){
            int tmp;
            cin >> tmp;
            s.push(tmp);
        }
        else if(str == "pop"){
            if(s.empty()){
                cout << -1 << "\n";
            }else {
                cout << s.top() << "\n";
                s.pop();
            }
        }
        else if(str == "size"){
            cout << s.size() << "\n";
        }
        else if(str== "empty"){
            cout << ((s.empty())? 1:0) << "\n";
        }
        else if(str == "top"){
            if(s.empty()){
                cout << -1 << "\n";
            }else cout << s.top() << "\n";
        }
    }

    return 0;
}

File: DONGUK/WEEK 5/10828.py
import sys
input = sys.stdin.readline

n = int(input())

stack = []

for _ in range(n):
    s = input().rstrip().split()
    
    if len(s) > 1:
        if s[0] == 'push':
            stack.append(s[1])

    else:
        if s[0] == 'top':
            if len(stack) == 0:
                print(-1)
            else:
                print(stack[-1])
        
        elif s[0] == 'pop':
            if len(stack) == 0:
                print(-1)
            else:
                print(stack.pop())
                
        elif s[0] == 'size':
            print(len(stack))
        
        elif s[0] == 'empty':
            if len(stack) == 0:
                print(1)
            
            else:
                print(0)
